### ISSUE TITLE
Fift interpreter error in asm code

### ISSUE THREAD

**Issue by Kyborg2011** [ton-blockchain/ton#102]
Hello!

I have get some errors:
```
$ func -o smart-contract.fif lib/stdlib.fc smart-contract.fc
$ fift -s new-wallet.fif 0 10 3
Creating new multi-signature wallet in workchain 0 
[ 1][t 0][1570491132.555057049][words.cpp:2784] smart-contract.fif:188: s18: -?
[ 1][t 0][1570491132.555223227][words.cpp:2784] new-wallet.fif:20: include: error interpreting included file `smart-contract.fif` : smart-contract.fif:188: s18: -?
[ 1][t 0][1570491132.555312157][fift-main.cpp:198]      Error interpreting file `new-wallet.fif`: error interpreting included file `new-wallet.fif` : new-wallet.fif:20: include: error interpreting included file `smart-contract.fif` : smart-contract.fif:188: s18: -?
```

How to understand this "**s18: -?**" ? Is it a bug or I am do something wrong?

**Comment by pfletcherhill** [comment]
Usually when I get an error like that it means that the word `s18` is not defined. Are you calling `s18` inside a Fift assembler block (ie `"Asm.fif" include <{ s1 s2 XCHG OVER }>s`)?

**Comment by Kyborg2011** [comment]
> Usually when I get an error like that it means that the word `s18` is not defined. Are you calling `s18` inside a Fift assembler block (ie `"Asm.fif" include <{ s18 }>s`)?

No, I have found a way to solve my code...
I think it is a bug in func compiler! Because, please, look at fiftbase.pdf, 7.2 on page 60:

> • s0. . . s15 ( – s), words similar to s1, but pushing the Slice represent- ing other “stack registers” of TVM. Notice that s16...s255 must be accessed using the word s().
> • s() (x – s), takes an Integer argument 0 ≤ x ≤ 255 and returns a special Slice used by the Fift assembler to represent “stack register” s(x).

So, in Fift assembler, as I see, "s16", "s17", etc. can't work properly, but instead of this, we must use "16 s()", "17 s()". So, func compiler have bring to me some code:
```
s1 s17 XCHG
s0 s17 XCHG
```
It doesn't work. I have changed it to:
```
s1 17 s() XCHG
s0 17 s() XCHG
```
And everything become fine!

I think it is a truly bug in func compiler...

**Comment by ton-blockchain** [comment]
Yes, it was a bug. It has been fixed in commit 38c3e39066a9c3b8a62b2f4cad54df4dc6307360.

**Comment by pfletcherhill** [comment]
Ah yeah, good call. What func code generates `s1 s17 XCHG`? Do you have a test case showing the bug?

**Comment by Kyborg2011** [comment]
> Yes, it was a bug. It has been fixed in commit [38c3e39](https://github.com/ton-blockchain/ton/commit/38c3e39066a9c3b8a62b2f4cad54df4dc6307360).

Thank you very much!
